bvm/ballerina-runtime/src/main/java/io/ballerina/runtime/internal/values/ArrayValueImpl.java

```diff
@@ -556,7 +556,7 @@ public void addRefValue(long index, Object value) {
             case TypeTags.UNSIGNED16_INT_TAG:
             case TypeTags.UNSIGNED8_INT_TAG:
                 prepareForAdd(index, value, type, intValues.length);
-                this.intValues[(int) index] = (Long) value;
+                this.intValues[(int) index] = Long.parseLong(String.valueOf(value));
```

Comment:
Why is this needed?

Reply:
Removed d759431e4a477acf23fed76b56fef739bb264967